src/System.Runtime/tests/System/String.cs

```diff
@@ -10,32 +10,20 @@
 public static unsafe class StringTests
 {
     private const string c_SoftHyphen = "\u00AD";
-
-    [Fact]
-    public static void TestWithEmptyString()
-    {
-        Assert.True("".EndsWith(""));
-        Assert.False("".EndsWith("Foo"));
-    }
-
-    [Fact]
-    public static void TestLastIndexOfWithEmptyString()
-    {
-        string s = "Dill Guv Dill Guv Dill";
-        int lastIndex = s.LastIndexOf("", StringComparison.OrdinalIgnoreCase);
-        Assert.Equal(s.Length - 1, lastIndex);
-    }
-
-    [Fact]
-    public static void TestCtor()
+    
+    [Theory]
+    [InlineData((string)null, true)]
+    [InlineData((string)"abc", false)]
+    public static void TestCtor(string s, bool isNull)
     {
-        string s = null;
-
-        Assert.Null(s);
-
-        s = "abc";
-
-        Assert.NotNull(s);
+        if (isNull)
+        {
+            Assert.Null(s);
+        }
+        else
+        {
+            Assert.NotNull(s);
+        }
```

Comment:
This looks unnecessarily complex.  The original test that did ```string s = null; Assert.Null(s);``` isn't really a test of string at all, just of reference types and xunit, so I'd cut that. Even the second test that does ```string s = "abc"; Assert.NotNull(s);``` isn't particularly meaningful.  I'd suggest just cutting this whole thing.